### ISSUE TITLE
Global and component focus state color

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#6458]
### Description

Our focus `outline` is set to `"2px solid --calcite-ui-brand`" by default, but we should ensure the focus is exposed across all components and devs can modify to fit their need.

We will also need to document to provide context and process to designers and developers to support WCAG 2.2's upcoming level A requirement with [2.4.7: Focus Visible (Level A)](https://w3c.github.io/wcag/understanding/focus-visible.html).

cc @macandcheese 

### Acceptance Criteria

Exposed and documented CSS variable which sets the focus color default to `--calcite-ui-brand`, which can be modified.

Devs should also be able to edit the CSS variable to a color of their choice, for a specific component, or across all components.

### Relevant Info

We should also complete an audit on our hitbox focus across components to improve focus throughout our components - refer to #4633

### Which Component

All components

### Example Use Case

Devs who currently don't like the styling can turn off focus by setting the `outline` to `"0"`.

We should provide a way to support devs who wish to change the focus color, or as a last resort set the color as `"transparent"` so the focus state remains to ensure we're meeting and supporting WCAG 2.2 out of the box, and documenting how to change depending on branding needs.

### Esri team

Calcite (dev)

**monday.com sync:** #9366659587

**Comment by geospatialem** [comment]
https://github.com/Esri/calcite-components/pull/6782 fulfilled the color request.

The above request should be re-scoped to determine if we should also expose a CSS prop allowing devs to adjust the outline width.

**Comment by geospatialem** [comment]
Consider the offset inverse with offset width and if the offset type (such as dashed) could be exposed.

**Comment by jcfranco** [comment]
For future reference, we can apply the same pattern used for the global focus color to this and similar tokens (see https://github.com/Esri/calcite-design-system/pull/11960 for more details).

**Comment by DintaMel** [comment]
@matgalla Can you take a look to see if this is already done?

**Comment by geospatialem** [comment]
> [@matgalla](https://github.com/matgalla) Can you take a look to see if this is already done?

Chatted with @matgalla and @jcfranco and the above has been achieved with different requirements from https://github.com/Esri/calcite-design-system/pull/11960 and https://github.com/Esri/calcite-design-system/pull/13398. Closing the above out as a duplicate.
